connectors/sources/s3.py

```diff
@@ -18,13 +18,27 @@
 from contextlib import asynccontextmanager
 from connectors.logger import logger, set_extra_logger
 from connectors.source import BaseDataSource
+from connectors.utils import get_base64_value
+
 
 SUPPORTED_CONTENT_TYPE = [
     "text/plain",
 ]
-SUPPORTED_FILETYPE = [".py", ".rst", ".rb", ".sh", ".md", ".txt"]
-ONE_MEGA = 1048576
+SUPPORTED_FILETYPE = [
```

Comment:
Tika supports a ton of file types, but the Attachment Processor doesn't (it'll just return the empty string for a variety of types). [Workplace search's list of extensions](https://github.com/elastic/ent-search/blob/main/connectors/lib/connectors/subextractor/full_text.rb#L6-L22) is probably the right place to start.

```
        doc
        docx
        html
        odt
        one
        md
        markdown
        paper
        pdf
        ppt
        pptx
        rtf
        txt
        xls
        xlsx
```

Reply:
@seanstory @tarekziade I tried to extract the content for .one filetype but it seems that it's not extracting the content. can you please confirm once that .one is supported by attachment processor or not?

Reply:
@akanshi-crest do you get an error?

Reply:
No I didn't get any error but unable to index the body field for the same.

Reply:
> unable to index the body field for the same

not sure what you mean. Is `_attachement` getting removed from the doc?

Reply:
yes `_attachement` is removed from the doc but in elastic we are not getting body field for `.one` document. Attaching the screenshot of the elastic document
![Screenshot (9)](https://user-images.githubusercontent.com/95739270/205862118-779c0fdd-c344-45c8-adee-df71ee49c107.png)

Reply:
I noticed that when the ingest attachment process picks a file, we don't have a lot of logs to see what is happening. I did not find any options for this in Elasticsearch. The only thing I see is a couple of APIs to check that it was really called. Before I investigate, @seanstory do you know if there's a way to log the ingest pipeline activity and get that info?

Reply:
> There is not. It's a downside to the ingest pipelines, that you can't really do logging.

Reply:
Our pipelines are using `onFailure`. The processor has a clause like:
```
"on_failure": [
        {
          "append": {
            "description": "Record error information",
            "field": "_ingestion_errors",
            "value": "Processor 'attachment' in pipeline '{{ _ingest.on_failure_pipeline }}' failed with message '{{ _ingest.on_failure_message }}'"
          }
        }
      ]
```

If there's no `_ingestion_errors` field on the output document, then there wasn't a failure, and the attachment processor just didn't output a `body`. That can happen. It's fine. It just means there wasn't searchable text on that file as far as Tika could find.

Reply:
Semi-related, I made a test using a Python file, which is not recognized by Tika and is silently ignored.

One trick I used to make sure it was indexed is to pass a filename to the ingest attachment that finishes with `.txt` and it worked.

That makes me wonder if we should not keep a list of file types we know are text-based, and in that case send the content of the file like we initially did here, or trick Tika to think it's `.txt`

WDYT @seanstory ?

Reply:
Can you clarify, was the python file dropped by Tika, or dropped by us before sending to the Attachment Processor?

IDK exactly how the attachment processor works, but Tika shouldn't care about file extensions, and should be able to "see through" mis-named files. I would have expected python files to be treated like text by tika without any shenanigans.

Reply:
I sent it to the ingest attachment pipeline in base64, and that did not work. I would assume Tika can't guess the mimetype on non-binary files because it can't look for hints in the binary structure. 

Now, you can give the pipeline a field with the filename and tell tika to guess the mimetype using the extension.
Same thing, did not work for `.py`

Now here's the trick: if I lie and replace `py` by `txt` in tha field, and push the file content, it will get ingested correctly.

Reply:
it's the `resource_name` field see https://www.elastic.co/guide/en/elasticsearch/reference/8.5/attachment.html